pkg/cluster/operation/action.go

```diff
@@ -452,7 +453,13 @@ func EnableMonitored(
 }
 
 // StartComponent start the instances.
-func StartComponent(ctx context.Context, instances []spec.Instance, options Options, tlsCfg *tls.Config) error {
+func StartComponent(
+	ctx context.Context,
+	instances []spec.Instance,
+	options Options,
+	tlsCfg *tls.Config,
+	finalCluster spec.Topology,
```

Comment:
How about passing a PD client object to the function, I think the `tlsCfg` could also be removed that way.

Reply:
Since `Instance` is an abstraction interface used by both TiDB and DM, while PD only exists in the TiDB scenario. IMHO it is not very good to pass PD client as parameters, or we can abstract a `MetaClient` interface, whose responsibility is for these metadata operations by the client, but the kinds of metadata operation in TIDB and DM are so different, seems hard to abstract.

Reply:
Then what about put the base topology into ctx since we may access it anywhere? Insert it at here: https://github.com/pingcap/tiup/blob/master/pkg/cluster/manager/scale_out.go#L151

Reply:
sounds good to me.